Parking-Zone/Services/ParkingSlotService.cs

```diff
@@ -35,5 +39,20 @@ public bool IsSlotFree(ParkingSlot slot, DateTime startTime, int duration)
             => !slot.Reservations.Any(reservation =>
                 reservation.StartTime <= startTime && startTime < reservation.StartTime.AddHours(reservation.Duration) ||
                 reservation.StartTime > startTime && startTime.AddHours(duration) > reservation.StartTime);
+
+        public IEnumerable<ParkingSlot> Filter(FilterSlotsQuery slotVM)
+        {
+            var slots = GetByParkingZoneId(slotVM.ZoneId);
+
+            if (slotVM.Category == 0)
+            {
+                if (slotVM.OnlyFree)
+                {
+                    return slots.Where(s => s.HasAnyActiveReservation != slotVM.OnlyFree);
+                }
+                return slots;
+            }
+            return slots.Where(s => s.Category == slotVM.Category && s.HasAnyActiveReservation != slotVM.OnlyFree);
+        }
```

Comment:
this method will not work correctly in some cases, for example when category != 0  va onlyFree = false